internal/cli/iam/organizations/create_atlas.go

```diff
@@ -0,0 +1,158 @@
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/prerun"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+var createAtlasTemplate = "Organization '{{.Organization.ID}}' created.\n"
+
+type CreateAtlasOpts struct {
+	cli.OutputOpts
+	name              string
+	ownerID           string
+	apiKeyDescription string
+	apiKeyRole        []string
+	store             store.AtlasOrganizationCreator
+}
+
+func (opts *CreateAtlasOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *CreateAtlasOpts) Run() error {
+	o := &mongodbatlas.CreateOrganizationRequest{
+		Name: opts.name,
+	}
+	if opts.ownerID != "" {
+		o.OrgOwnerID = &opts.ownerID
+	}
+	if len(opts.apiKeyRole) > 0 {
+		o.APIKey = &mongodbatlas.APIKeyInput{}
+		o.APIKey.Roles = opts.apiKeyRole
+		o.APIKey.Desc = opts.apiKeyDescription
+	}
+
+	r, err := opts.store.CreateAtlasOrganization(o)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *CreateAtlasOpts) validateAPIKeyRequirements() error {
+	required := make([]string, 0)
+	if len(opts.apiKeyRole) == 0 {
+		required = append(required, flag.APIKeyRole)
+	}
+	if opts.ownerID == "" {
+		required = append(required, flag.OwnerID)
+	}
+	if len(required) > 0 {
+		return fmt.Errorf(
+			"%s are required when using API keys to authenticate",
+			strings.Join(required, ", "),
+		)
+	}
+	return nil
+}
+
+func (opts *CreateAtlasOpts) validateOAuthRequirements() error {
+	disallowed := make([]string, 0)
+	if len(opts.apiKeyRole) > 0 {
+		disallowed = append(disallowed, flag.APIKeyRole)
+	}
+	if opts.ownerID != "" {
+		disallowed = append(disallowed, flag.OwnerID)
+	}
+	if len(disallowed) > 0 {
+		return fmt.Errorf(
+			"%s are not allowed when using account to authenticate",
+			strings.Join(disallowed, ", "),
+		)
+	}
+	return nil
+}
+
+func (opts *CreateAtlasOpts) validateAuthType() error {
+	switch config.AuthType() {
+	case config.APIKeys:
+		return opts.validateAPIKeyRequirements()
+	case config.OAuth:
+		return opts.validateOAuthRequirements()
+	case config.NotLoggedIn:
+		return opts.validateOAuthRequirements()
+	default:
+		return nil
+	}
+}
+
+// CreateAtlasBuilder atlas organization(s) create <name>.
+func CreateAtlasBuilder() *cobra.Command {
+	opts := new(CreateAtlasOpts)
+
+	cmd := &cobra.Command{
+		Use:   "create <name>",
+		Short: "Create an Ops Manager or Cloud Manager organization. This command is unavailable for Atlas.",
+		Args:  require.ExactArgs(1),
```

Comment:
Question: Can any role this command? If not, can we add the required role as Long text here (replacing Organization Project Creator with whatever the required role is)?
```suggestion
		Long: fmt.Sprintf(usage.RequiredRole, "Organization Project Creator"),
		Args:  require.ExactArgs(1),
```

Reply:
Complicated, this command has two behaviors, either using API keys, it will need to be org owner and belong to an org with cross org billing enabled, or when using `atlas login` in which case any role

Reply:
Ah ok, let's hold off on the above suggestion then. Thanks!